zio-http/src/main/scala/zhttp/http/middleware/HttpMiddlewares.scala

```diff
@@ -0,0 +1,196 @@
+    patch((_, _) => Patch.addHeader(headers))
+
+  def addCookieM[R, E](cookie: ZIO[R, E, Cookie]): HttpMiddleware[R, E] =
+    patchZIO((_, _) => cookie.mapBoth(Option(_), c => Patch.addHeader(Headers.setCookie(c))))
+
+  /**
+   * Add log status, method, url and time taken from req to res
+   */
+  def debug: HttpMiddleware[Console with Clock, IOException] =
+    makeResponseZIO((method, url, _) => zio.clock.nanoTime.map(start => (method, url, start))) {
+      case (status, _, (method, url, start)) =>
+        for {
+          end <- clock.nanoTime
+          _   <- console
+            .putStrLn(s"${status.asJava.code()} ${method} ${url.asString} ${(end - start) / 1000000}ms")
+            .mapError(Option(_))
+        } yield Patch.empty
+    }
+
+  /**
+   * Creates a new middleware using transformation functions
+   */
+  def makeResponse[S](req: (Method, URL, Headers) => S): PartialResponseMake[S] = PartialResponseMake(req)
+
+  /**
+   * Creates a new middleware using effectful transformation functions
+   */
+  def makeResponseZIO[R, E, S](req: (Method, URL, Headers) => ZIO[R, Option[E], S]): PartialResponseMakeZIO[R, E, S] =
+    PartialResponseMakeZIO(req)
+
+  /**
+   * Creates a middleware that produces a Patch for the Response
+   */
+  def patch[R, E](f: (Status, Headers) => Patch): HttpMiddleware[R, E] =
+    makeResponse((_, _, _) => ())((status, headers, _) => f(status, headers))
+
+  /**
+   * Creates a middleware that produces a Patch for the Response effectfully.
+   */
+  def patchZIO[R, E](f: (Status, Headers) => ZIO[R, Option[E], Patch]): HttpMiddleware[R, E] =
+    makeResponseZIO((_, _, _) => ZIO.unit)((status, headers, _) => f(status, headers))
+
+  /**
+   * Removes the header by name
+   */
+  def removeHeader(name: String): HttpMiddleware[Any, Nothing] =
+    patch((_, _) => Patch.removeHeaders(List(name)))
+
+  /**
+   * Runs the effect after the response is produced
+   */
+  def runAfter[R, E](effect: ZIO[R, E, Any]): HttpMiddleware[R, E] =
+    patchZIO((_, _) => effect.mapBoth(Option(_), _ => Patch.empty))
+
+  /**
+   * Runs the effect before the request is passed on to the HttpApp on which the middleware is applied.
+   */
+  def runBefore[R, E](effect: ZIO[R, E, Any]): HttpMiddleware[R, E] =
+    makeResponseZIO((_, _, _) => effect.mapError(Option(_)).unit)((_, _, _) => UIO(Patch.empty))
+
+  /**
+   * Creates a new middleware that always sets the response status to the provided value
+   */
+  def status(status: Status): HttpMiddleware[Any, Nothing] = patch((_, _) => Patch.setStatus(status))
+
+  /**
+   * Creates a new constants middleware that always executes the app provided, independent of where the middleware is
+   * applied
+   */
+  def fromApp[R, E](app: HttpApp[R, E]): HttpMiddleware[R, E] = Middleware.fromHttp(app)
+
+  /**
+   * Times out the application with a 408 status code.
+   */
+  def timeout(duration: Duration): HttpMiddleware[Clock, Nothing] =
```

Comment:
May be there should be a `timeout` middleware also ?

Reply:
https://github.com/dream11/zio-http/pull/797